library/sha512.c

```diff
@@ -445,10 +453,16 @@ static const unsigned char sha512_test_sum[6][64] =
 int mbedtls_sha512_self_test( int verbose )
 {
     int i, j, k, buflen, ret = 0;
-    unsigned char buf[1024];
+    unsigned char *buf;
     unsigned char sha512sum[64];
     mbedtls_sha512_context ctx;
 
+    buf = mbedtls_calloc(1024, sizeof(unsigned char));
```

Comment:
Style: we should have spaces after and before the parentheses belonging to the function.